guides/developer/service-overviews/orca.md

```diff
@@ -0,0 +1,48 @@
+
+## Runtime
+
+Orca uses a distributed queue library, [Keiko](http://github.com/spinnaker/keiko), to manage its work. Keiko uses atomic messages to progress work through its lifecycle and allows Orca to first be resilient to node failure, as well as spread load evenly across the deployment making Orca easier to scale and operate.
+
+Keiko itself is an abstraction around a delay queue, meaning it can deliver **Messages** immediately or at specific times in the future, as well as reschedule messages that have already been added to the queue for a new delivery time. This delay queue functionality is pivotal to Orca's performance and operational characteristics when dealing with long running operations.
+
+A Message is a fine-grained unit of work. As an example, a `StartStage` message will verify that all upstream branches are complete and then perform any logic associated with preparing for the stage's execution, it won't actually perform the stage's core actions. A `RunTask` is really the only message that actually performs stage work that a user would be familiar with.
+
+Some message types are redeliverable and duplicated, whereas others are not. For example, in a pipeline that has two concurrent stage branches that do not join at the end, there will be two `CompleteExecution` messages sent. This is fine, because the `CompleteExecutionHandler` has logic to verify that indeed all stages are in a completed state before marking the entire execution as complete, whether it ultimately be a failure or success.
+
+For every message, there is a single correlated `Handler` that contains all of the logic for processing this `Message`. A Message's contents are small and the bare minimum of information to process the request: For instance, an Execution is referenced by ID rather than containing its entire state. Orca builds atop these foundational Handlers in the form of `StageDefinitionBuilder` and other classes to build the DAG and implement actual Spinnaker orchestration logic.
+
+The queue can be categorized into two parts: The **QueueProcessor**, and its Handlers. Orca uses a single thread to run `QueueProcessor`, which polls the oldest messages "ready" off the queue. A ready message is one whose delivery time is now or in the past: It is normal to have many more messages in the queue than those that are actually ready. 
+
+A separate worker thread pool is used for Handlers. Orca’s ability to scale to process its work is dependent on availability of threads in the worker pool which can be tuned either by adjusting the worker thread pool size, or by increasing the number of Orca instances: Ignoring downstream bottlenecks, Orca can be scaled out horizontally to meet the work demands Spinnaker is given.
```

Comment:
I took a shot at simplifying this paragraph, Rob. Can you verify I didn't break it?

"A separate worker thread pool is used for Handlers.  Orca depends on threads in that pool in order to scale sufficiently. You can tune threads either by adjusting the pool size or by increasing the number of Orca instances. By ignoring downstream bottlenecks Orca can scale horizontally to meet Spinnaker work demand."